MekHQ/src/mekhq/campaign/universe/StarType.java

```diff
@@ -0,0 +1,422 @@
+    public static final String LUM_III_EVOLVED = "II/III";
+    public static final String LUM_III = "III";
+    public static final String LUM_IV_EVOLVED = "III/IV";
+    public static final String LUM_IV = "IV";
+    public static final String LUM_V_EVOLVED = "IV/V";
+    public static final String LUM_V = "V";
+    // typically used as a prefix "sd", not as a suffix
+    public static final String LUM_VI = "VI";
+    // typically used as a prefix "esd", not as a suffix
+    public static final String LUM_VI_PLUS = "VI+";
+    // always used as class designation "D", never as a suffix
+    public static final String LUM_VII = "VII";
+
+    private int spectralClass;
+    private double subtype;
+    private String luminosity;
+
+    public StarType() {
+        // empty constructor
+    }
+
+    public StarType(int spectraClass, double subtype, String luminosity) {
+        this.spectralClass = spectraClass;
+        this.subtype = subtype;
+        this.luminosity = luminosity;
+    }
+
+    public int getSpectralClass() {
+        return spectralClass;
+    }
+
+    public void setSpectralClass(int c) {
+        this.spectralClass = c;
+    }
+
+    public double getSubtype() {
+        return subtype;
+    }
+
+    public void setSubType(double d) {
+        this.subtype = d;
+    }
+
+    public String getLuminosity() {
+        return luminosity;
+    }
+
+    public void setLuminosity(String s) {
+        this.luminosity = s;
+    }
+
+    @Override
+    public String toString() {
+        if (spectralClass == StarType.SPECTRAL_Q) {
+            return (null != luminosity) ? "Q" + luminosity : "Q";
+        }
+
+        // Formatting subtype value up to two decimal points, if needed
+        int subtypeValue = MathUtility.clamp((int) Math.round(subtype * 100d), 0, 999);
+
+        String subtypeFormat = "%.2f";
+        if (subtypeValue % 100 == 0) {
+            subtypeFormat = "%.0f";
+        } else if (subtypeValue % 10 == 0) {
+            subtypeFormat = "%.1f";
+        }
+
+        if (null != luminosity && luminosity.equals(StarType.LUM_VI)) {
+            // subdwarfs
+            return "sd" + getSpectralClassName(spectralClass) + String.format(subtypeFormat, subtypeValue / 100.0);
+        } else if (null != luminosity && luminosity.equals(StarType.LUM_VI_PLUS)) {
+            // extreme subdwarfs
+            return "esd" + getSpectralClassName(spectralClass) + String.format(subtypeFormat, subtypeValue / 100.0);
+        } else if (null != luminosity && luminosity.equals(StarType.LUM_VII)) {
+            // white dwarfs
+            return String.format(Locale.ROOT, "D" + subtypeFormat, subtypeValue / 100.0);
+        } else {
+            // main class
+            return String.format(Locale.ROOT, "%s" + subtypeFormat + "%s",
+                getSpectralClassName(spectralClass),
+                subtypeValue / 100.0, (null != luminosity ? luminosity : StarType.LUM_V));
+        }
```

Comment:
This can be made more legible with an initial escape with a `null != luminosity` check and return whatever the default is then run a switch for the other 3 followed by the default WITH luminosity.